lib/deploy_server_creator.rb

```diff
@@ -88,25 +80,31 @@ def self.create(auth_token: nil, repositories: [])
           info = deployer.deploy_info
           return 200, { message: info }.to_json
         rescue IOError => error
-          EventLog.log(
-            request_id, ['user_agent', request.env['HTTP_USER_AGENT'].to_s]
-          )
-          EventLog.log request_id, error
+          log request_id, ['user_agent', request.env['HTTP_USER_AGENT'].to_s]
+          log request_id, error
           EventLog.flush request_id
           return 400, { error: error.message }.to_json
         end
       end
 
       get '/:repository_name/tags' do
-        repository = repositories.find do |repo_candidate|
-          repo_candidate.name == params['repository_name']
-        end
+        repository = find_repository(params)
 
         return not_found if repository.nil?
 
         tags = RepoTags.new(repository, 'master')
         return 200, { count: tags.count, names: tags.names }.to_json
       end
+
+      def find_repository(params)
+        repositories.find do |repo_candidate|
+          repo_candidate.name == params['repository_name']
+        end
+      end
+
+      def log(request_id, message)
```

Comment:
i like the direct usage of EventLog.log better, cause the dependency to EventLog is visible in every method.

Reply:
resolved